File: src/Envirotrax.App/envirotrax.app.client/src/app/shared/models/csi/csi-submission-create-view-model.ts
import { State } from '../lookup/state';
import { PropertyType } from '../../enums/property-type.enum';
import { CsiInspectionReason } from '../../enums/csi-inspection-reason.enum';

export interface CsiSubmissionCreateViewModel {
    siteId: number;
    site: CsiSiteData;
    inspector: CsiInspectorData;
    csiLicense?: CsiLicenseData;
    availableWaterSuppliers: CsiWaterSupplierOption[];
    defaultWaterSupplierId?: number;
    availableCsiAccounts: CsiAccountOption[];
    defaultCsiAccountUserId: number;
}

export interface CsiAccountOption {
    userId: number;
    contactName?: string;
    jobTitle?: string;
    emailAddress?: string;
    csiLicense?: CsiLicenseData;
}

export interface CsiSiteData {
    id: number;
    accountNumber?: string;
    businessName?: string;
    propertyType?: PropertyType;
    streetNumber?: string;
    streetName?: string;
    propertyNumber?: string;
    city?: string;
    state?: State;
    zipCode?: string;
    mailingCompanyName?: string;
    mailingContactName?: string;
    mailingStreetNumber?: string;
    mailingStreetName?: string;
    mailingNumber?: string;
    mailingCity?: string;
    mailingState?: State;
    mailingZipCode?: string;
    mailingPhoneNumber?: string;
    mailingEmailAddress?: string;
}

export interface CsiInspectorData {
    companyName?: string;
    contactName?: string;
    jobTitle?: string;
    address?: string;
    city?: string;
    state?: string;
    zipCode?: string;
    phoneNumber?: string;
    faxNumber?: string;
    emailAddress?: string;
}

export interface CsiLicenseData {
    licenseNumber?: string;
    licenseTypeName?: string;
    expirationDate?: string;
    isValid: boolean;
}

export interface CsiWaterSupplierOption {
    id: number;
    name: string;
    pwsId?: string;
    address?: string;
    city?: string;
    state?: string;
    zipCode?: string;
    phoneNumber?: string;
    contactName?: string;
    emailAddress?: string;
}

export interface CsiSubmissionSaveRequest {
    siteId: number;
    waterSupplierId: number;
    selectedCsiAccountUserId: number;
    inspectionDate?: string;
    reasonForInspection: CsiInspectionReason;
    compliance1: boolean;
    compliance2: boolean;
    compliance3: boolean;
    compliance4: boolean;
    compliance5: boolean;
    compliance6: boolean;
    materialServiceLineLead: boolean;
    materialServiceLineCopper: boolean;
    materialServiceLinePVC: boolean;
    materialServiceLineOther: boolean;
    materialServiceLineOtherDescription?: string;
    materialSolderLead: boolean;
    materialSolderLeadFree: boolean;
    materialSolderSolventWeld: boolean;
    materialSolderOther: boolean;
    materialSolderOtherDescription?: string;
    comments?: string;
}

File: src/Envirotrax.App/Envirotrax.App.Server/Domain/DataTransferObjects/Csi/CsiSubmissionCreateViewModel.cs
using Envirotrax.App.Server.Data.Models.Sites;
using Envirotrax.App.Server.Domain.DataTransferObjects.Lookup;

namespace Envirotrax.App.Server.Domain.DataTransferObjects.Csi;

public class CsiSubmissionCreateViewModel
{
    public int SiteId { get; set; }
    public CsiSiteDataDto Site { get; set; } = null!;
    public CsiInspectorDataDto Inspector { get; set; } = null!;
    public CsiLicenseDataDto? CsiLicense { get; set; }
    public List<CsiWaterSupplierOptionDto> AvailableWaterSuppliers { get; set; } = [];
    public int? DefaultWaterSupplierId { get; set; }
    public List<CsiAccountOptionDto> AvailableCsiAccounts { get; set; } = [];
    public int DefaultCsiAccountUserId { get; set; }
}

public class CsiSiteDataDto
{
    public int Id { get; set; }
    public string? AccountNumber { get; set; }
    public string? BusinessName { get; set; }
    public PropertyType PropertyType { get; set; }
    public string? StreetNumber { get; set; }
    public string? StreetName { get; set; }
    public string? PropertyNumber { get; set; }
    public string? City { get; set; }
    public ReferencedStateDto? State { get; set; }
    public string? ZipCode { get; set; }
    public string? MailingCompanyName { get; set; }
    public string? MailingContactName { get; set; }
    public string? MailingStreetNumber { get; set; }
    public string? MailingStreetName { get; set; }
    public string? MailingNumber { get; set; }
    public string? MailingCity { get; set; }
    public ReferencedStateDto? MailingState { get; set; }
    public string? MailingZipCode { get; set; }
    public string? MailingPhoneNumber { get; set; }
    public string? MailingEmailAddress { get; set; }
}

public class CsiInspectorDataDto
{
    public string? CompanyName { get; set; }
    public string? ContactName { get; set; }
    public string? JobTitle { get; set; }
    public string? Address { get; set; }
    public string? City { get; set; }
    public string? State { get; set; }
    public string? ZipCode { get; set; }
    public string? PhoneNumber { get; set; }
    public string? FaxNumber { get; set; }
    public string? EmailAddress { get; set; }
}

public class CsiLicenseDataDto
{
    public string? LicenseNumber { get; set; }
    public string? LicenseTypeName { get; set; }
    public DateTime? ExpirationDate { get; set; }
    public bool IsValid { get; set; }
}

public class CsiAccountOptionDto
{
    public int UserId { get; set; }
    public string? ContactName { get; set; }
    public string? JobTitle { get; set; }
    public string? EmailAddress { get; set; }
    public CsiLicenseDataDto? CsiLicense { get; set; }
}

public class CsiWaterSupplierOptionDto
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public string? PwsId { get; set; }
    public string? Address { get; set; }
    public string? City { get; set; }
    public string? State { get; set; }
    public string? ZipCode { get; set; }
    public string? PhoneNumber { get; set; }
    public string? ContactName { get; set; }
    public string? EmailAddress { get; set; }
}
